SparkPost/2web2ui
# FE-617: List page pattern

# What I Did
I built a reusable `ListPage` component to satisfy to main use cases of our list pages. I created a 2nd user list page to demo the new component.

The goal is to abstract and simplify common patterns to make them easier to reuse and maintain, and also to reduce overall codebase size. We have 12 basic list pages, each with their own basic layout and state management, and matching test suites.

# Testing
 - Check out the `ListPage` stories: `npm run storybook`
 - Try the re-implemented users list page by editing `routes.js` to point `/account/users` to `users.ListPage2`.

# To Do
 - [x] Test `pages/users/ListPage2.js`
 - [ ] Hide item actions list when not available
 - [ ] Disable delete dialog during delete
 - [x] Change PR base to master once FE-615 is merged down
